ansible/playbooks/roles/upgrade/tasks/kubernetes.yml

```diff
@@ -1,25 +1,67 @@
 ---
+# During HA control plane upgrade server address in kubeconfig is switched to local for
+#   * compatibility between client and server versions
+#   * identifying correct server version
+
+- name: k8s/master | Switch apiserver address to local
+  include_tasks: kubernetes/utils/set-local-apiserver.yml # sets kubectl_context_cluster
+  when:
+    - groups.kubernetes_master | length > 1
+    - inventory_hostname in groups.kubernetes_master
+
 - name: Wait for kube-apiserver and get cluster version
-  delegate_to: "{{ groups.kubernetes_master[0] }}"
+  delegate_to: >-
+    {{ inventory_hostname if inventory_hostname in groups.kubernetes_master else
+      groups.kubernetes_master[0] }}
   block:
     - name: k8s | Include wait-for-kube-apiserver.yml
       import_tasks: kubernetes/utils/wait-for-kube-apiserver.yml
 
     - name: k8s | Include get-cluster-version.yml
-      import_tasks: kubernetes/get-cluster-version.yml # sets cluster_version
+      import_tasks: kubernetes/get-cluster-version.yml
+
+    - name: Set initial_cluster_version fact # sets initial_cluster_version
+      set_fact:
+        initial_cluster_version: >-
+          {{ (kubectl_cluster_version.stdout | from_yaml).serverVersion.gitVersion }}
+
+    - name: k8s/kubelet-cm | Include set-cluster-version.yml
```

Comment:
`k8s/kubelet-cm` - is that name correct?

Reply:
That name comes from [ansible/playbooks/roles/upgrade/tasks/kubernetes/patch-kubelet-cm.yml](https://github.com/epiphany-platform/epiphany/pull/2766/files/8039af8429e265e518dd817ad02b362ccca93d0f#diff-ee268bede56a4d59831f4e37a661445606a5f6669f046921cc627e5b0bbd1b51) so it is my mistake of not changing it here. I will fix it